unittests/AST/ASTImporterTest.cpp

```diff
@@ -805,6 +805,30 @@ class _Mem_fn<_Res _Class::*> {
                          namespaceDecl()));
 }
 
+TEST_F(Fixture, ImportFunctionWithBackReferringParameter) {
+  Decl *From, *To;
+  std::tie(From, To) = getImportedDecl(
+    R"(
+template<typename _T>
+struct X {};
+
+void declToImport(int y,X<int> &x){}
+
+template<>
+struct X<int> {
+  void g(){
+    X<int> x;
+    declToImport(0,x);
+  }
+};
+    )",Lang_CXX, "", Lang_CXX);
+
+  MatchVerifier<Decl> Verifier;
+  auto Matcher = functionDecl(hasName("declToImport"),parameterCountIs(2),hasParameter(0,hasName("y")),hasParameter(1,hasName("x")),hasParameter(1,hasType(asString("X<int> &"))));
```

Comment:
Use `clang-format` please. LLVM/Clang is really strict about 80 line width and other formatting. https://llvm.org/docs/CodingStandards.html

Reply:
How to "squash" the commits? Reset locally (undo last 3 commits) and make a single commit out of it (after push what happens to the existing but removed remote commits)? "Merge commits" are not totally understandable from the documentation but it seems that there is an option (at pull request merge) to squash the commits.